### PR TITLE
Travis: use stable SOURCE_DATE_EPOCH for predictable calendar examples

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/awesomeWM/awesome/issues/2070

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,13 +12,14 @@ env:
...
     # Lua 5.3 isn't available in Ubuntu Trusty, so some magic below installs it.
-    - LUA=5.3 LUANAME=lua5.3 LUALIBRARY=/usr/lib/liblua.so DO_COVERAGE=codecov EXPORT_SDE=true
+    - LUA=5.3 LUANAME=lua5.3 LUALIBRARY=/usr/lib/liblua.so DO_COVERAGE=codecov
     # luajit: installed from source.
     - LUA=5.1 LUANAME=luajit-2.0 LUALIBRARY=/usr/lib/libluajit-5.1.so LUAROCKS_ARGS=--lua-suffix=jit-2.0.5 TEST_PREV_COMMITS=1
     # Note: luarocks does not work with Lua 5.0.
     - LUA=5.1 LUANAME=lua5.1 BUILD_IN_DIR=/tmp/awesome-build
     # Lua 5.2 with older lgi and screen size not divisible by 2.
-    - LUA=5.2 LUANAME=lua5.2 LGIVER=0.8.0 TESTS_SCREEN_SIZE=1921x1079 BUILD_APIDOC=true DO_CHECKQA=1 EMPTY_THEME_WHILE_LOADING=1
+    # SOURCE_DATE_EPOCH: used for stable dates during build.  Should be the unix timestamp of the last release (git log --pretty=format:%cd -n 1 --date=unix v4.2).
```
Comment (blueyed) [inline .travis.yml]: @psychon 
Added a comment.  Could be added to the how-to-release doc - where is that again?

Comment (psychon) [reply .travis.yml]: `awesome-www/doing_a_release.mdwn`

File: .travis.yml [hunk 2]
```diff
@@ -12,13 +12,13 @@ env:
...
     # Lua 5.3 isn't available in Ubuntu Trusty, so some magic below installs it.
-    - LUA=5.3 LUANAME=lua5.3 LUALIBRARY=/usr/lib/liblua.so DO_COVERAGE=codecov EXPORT_SDE=true
+    - LUA=5.3 LUANAME=lua5.3 LUALIBRARY=/usr/lib/liblua.so DO_COVERAGE=codecov
     # luajit: installed from source.
     - LUA=5.1 LUANAME=luajit-2.0 LUALIBRARY=/usr/lib/libluajit-5.1.so LUAROCKS_ARGS=--lua-suffix=jit-2.0.5 TEST_PREV_COMMITS=1
     # Note: luarocks does not work with Lua 5.0.
     - LUA=5.1 LUANAME=lua5.1 BUILD_IN_DIR=/tmp/awesome-build
     # Lua 5.2 with older lgi and screen size not divisible by 2.
-    - LUA=5.2 LUANAME=lua5.2 LGIVER=0.8.0 TESTS_SCREEN_SIZE=1921x1079 BUILD_APIDOC=true DO_CHECKQA=1 EMPTY_THEME_WHILE_LOADING=1
+    - LUA=5.2 LUANAME=lua5.2 LGIVER=0.8.0 TESTS_SCREEN_SIZE=1921x1079 BUILD_APIDOC=true DO_CHECKQA=1 EMPTY_THEME_WHILE_LOADING=1 SOURCE_DATE_EPOCH=1502906828
```
Comment (psychon) [inline .travis.yml]: What's so special about 16th of August? Could we take a date from before 2000 or after 2100 instead?

File: .travis.yml (hunk 1 above)
Comment (psychon) [reply .travis.yml]: But for releases it would just take the date of the release, I think. Shouldn't that be enough?

File: .travis.yml (hunk 2 above)
Comment (blueyed) [reply .travis.yml]: It was from the NEWS file.
Amended to use the last tag (`4.2`).

File: .travis.yml (hunk 1 above)
Comment (blueyed) [reply .travis.yml]: Yes, but it is fixed (I did not want to have a magic `git-log` that would look for the newest tag and use that.
So we should update the .travis.yml after a release - although it is not really important IMHO.
Would you prefer this to be automatic, i.e. using git-log magic?

Comment (psychon) [reply .travis.yml]: I mean: For releases, the doc is built by Travis on a build of the awesome-www repo. The date we use here is only used for the builds of the current devel version docs.

I suggest to use 1893456000 here. That's the output of `date -u  +%s -d '2030-01-01'`. That way, we can just leave it at that and do not have to update it.

Comment (blueyed) [reply .travis.yml]: Ok, amended.

**Pull request merged**